src/test/java/testsuite/integration/container/HikariCPIntegrationTest.java

```diff
@@ -0,0 +1,255 @@
+    Class.forName("software.aws.rds.jdbc.mysql.Driver");
+    log = LogFactory.getLogger(StandardLogger.class.getName(), Log.LOGGER_INSTANCE_NAME);
+
+    System.setProperty("com.zaxxer.hikari.blockUntilFilled", "true");
+  }
+
+  @AfterEach
+  public void teardown() {
+    data_source.close();
+  }
+
+  @BeforeEach
+  public void setUpTest() throws SQLException {
+    String writerEndpoint = (clusterTopology.size() > 0) ? clusterTopology.get(0) : "";
+
+    String jdbcUrl = URL_PREFIX + writerEndpoint + URL_SUFFIX;
+    log.logDebug("Writer endpoint: " + jdbcUrl);
+
+    final HikariConfig config = new HikariConfig();
+
+    config.setJdbcUrl(jdbcUrl);
+    config.setUsername(TEST_USERNAME);
+    config.setPassword(TEST_PASSWORD);
+    config.setMaximumPoolSize(3);
+    config.setReadOnly(true);
+    config.setExceptionOverrideClassName(HikariCPSQLException.class.getName());
+    config.setInitializationFailTimeout(75000);
+    config.setConnectionTimeout(1000);
+    config.addDataSourceProperty(PropertyKey.failoverTimeoutMs.getKeyName(), "5000");
+    config.addDataSourceProperty(PropertyKey.failoverReaderConnectTimeoutMs.getKeyName(), "1000");
+    config.addDataSourceProperty(PropertyKey.clusterInstanceHostPattern.getKeyName(), PROXIED_CLUSTER_TEMPLATE);
+    config.addDataSourceProperty(PropertyKey.failureDetectionTime.getKeyName(), "3000");
+    config.addDataSourceProperty(PropertyKey.failureDetectionInterval.getKeyName(), "1500");
+
+    data_source = new HikariDataSource(config);
+
+    final HikariPoolMXBean hikariPoolMXBean = data_source.getHikariPoolMXBean();
+
+    log.logDebug("Starting idle connections: " + hikariPoolMXBean.getIdleConnections());
+    log.logDebug("Starting active connections: " + hikariPoolMXBean.getActiveConnections());
+    log.logDebug("Starting total connections: " + hikariPoolMXBean.getTotalConnections());
+  }
+
+  /**
+   * After getting successful connections from the pool, the cluster becomes unavailable
+   */
+  @Test
+  public void test_1_1_hikariCP_lost_connection() throws SQLException {
+    try (Connection conn = data_source.getConnection()) {
+      assertTrue(conn.isValid(5));
+
+      putDownAllInstances(true);
+
+      final SQLException exception = assertThrows(SQLException.class, () -> queryInstanceId(conn));
+      assertEquals("08001", exception.getSQLState());
+      assertFalse(conn.isValid(5));
+    }
+
+    assertThrows(SQLTransientConnectionException.class, () -> data_source.getConnection());
+  }
+
+  /**
+   * After getting a successful connection from the pool, the connected instance becomes unavailable and the
+   * connection fails over to another instance. A connection is then retrieved to check that connections
+   * to failed instances are not returned
+   */
+  @Test
+  public void test_1_2_hikariCP_get_dead_connection() throws SQLException {
+    putDownAllInstances(false);
+
+    List<String> currentClusterTopology = getTopologyEndpoints();
```

Comment:
is there a reason why you're not using the static list `clusterTopology`?
Similar question about `test_2_1_hikariCP_efm_failover()`